pkg/cluster/kubeclient/ready.go

```diff
@@ -40,14 +41,27 @@ func (u *kubeclient) WaitForReadyWorker(ctx context.Context, hostname string) er
 func (u *kubeclient) WaitForReadySyncPod(ctx context.Context) error {
 	return wait.PollImmediateUntil(10*time.Second,
 		func() (bool, error) {
-			_, err := u.client.CoreV1().
+			d, err := u.client.AppsV1().Deployments("kube-system").Get("sync", metav1.GetOptions{})
+			switch {
+			case errors.IsNotFound(err):
+				return false, nil
+			case err != nil:
+				return false, err
+			}
+
+			ready := ready.DeploymentIsReady(d)
+			if ready {
+				return true, nil
+			}
+
+			_, err = u.client.CoreV1().
 				Services("kube-system").
 				ProxyGet("", "sync", "", "/healthz/ready", nil).
 				DoRaw()
 
 			switch {
 			case err == nil:
-				return true, nil
+				return false, nil
```

Comment:
This switch will now never return `true` and is basically just differentiating between different failure modes... can we simplify it?

Reply:
yes, thanks, pushed